### DISCUSSION TITLE
User Message: missing parameter hostname] service/connect.go:816

### DISCUSSION THREAD

**Discussion by Ayayouille** [gravitational/teleport#42960] [Q&A]
Hello everyone, I need help!

I configured Teleport on a Windows, with Docker desktop.

I managed to configure my server. And I can access it.

But when I want to register a machine, for example one of my debian machines, I can't do it.

I have this message :

2024-06-14 11:02:53     runtime/asm_amd64.s:1650 runtime.goexit
2024-06-14 11:02:53 User Message: missing parameter hostname] service/connect.go:816
2024-06-14 11:02:53 2024-06-14T09:02:53Z WARN [PROC:1]    Sync rotation state cycle failed. pid:7.1 retry_interval:10s service/connect.go:712
![Capture d'écran 2024-06-14 110708](https://github.com/gravitational/teleport/assets/103437811/ca6fbefa-d669-461b-bf42-3aeec545bc7c)

My configuration file on my windows looks like this:
![conf_windows](https://github.com/gravitational/teleport/assets/103437811/79b8c934-2aa1-40e3-be75-147af8fd60b6)

And on my linux client I have this :
![conf_linux](https://github.com/gravitational/teleport/assets/103437811/98691569-8382-437c-bcaf-b1e9baabfa92)

Can you please help me?

If you need anything else, don't hesitate to ask.

**Comment by webvictim** [comment]
Can you share the full stack trace that appears in the logs before the message you posted?

**Comment by Ayayouille** [comment]
Hello, I apologize for the delay in replying. I'm sorry my virtual machine wasn't working. Over the weekend, someone must have made a mistake and the virtual machine wouldn't turn on...

I had to create a new one and reconfigure my docker, with teleport.

I tried to redo the same configuration, but alas, I'm tired, I must admit. And now I have a certificate problem : 

![certifcate](https://github.com/gravitational/teleport/assets/103437811/3d4c9f28-f9b7-48d5-bd79-0915691b90fb)

And I confess I don't know how to solve this problem right away.

**Comment by Ayayouille** [comment]
I'll work on it again when I have more time.

**Comment by webvictim** [comment]
The `missing parameter hostname` message generally appears when you've changed the `proxy_service.public_addr` or `teleport.nodename` entries in the config file after starting the cluster up.

This looks like a bug, which I've reported here: https://github.com/gravitational/teleport/issues/43856

It can also happen when using an IP in the config file rather than a DNS hostname.
